build/deploy/cockroach.sh

```diff
@@ -1,10 +1,232 @@
+process_init_files() {
+	for f in "$@"; do
+		case "$f" in
+			*.sh)
+				if [ -x "$f" ]; then
+					echo "$0: running $f"
+					"$f"
+				else
+					echo "$0: sourcing $f"
+					. "$f"
+				fi
+				;;
+			*.sql)    echo "$0: running $f"; run_sql_query -f "$f"; echo ;;
+			*.sql.gz) echo "$0: running $f"; gzip -cd "$f" | run_sql_query; echo ;;
+			*.sql.xz) echo "$0: running $f"; xz --decompress --stdout "$f" | run_sql_query; echo ;;
+			*)        echo "$0: ignoring $f" ;;
+		esac
+		echo
+	done
+}
+
+# setup_db is to create a default database, create a default user (with password if applicable),
+# and grant privileges to this user by running `cockroach sql` queries.
+setup_db() {
+    create_defaultdb $COCKROACH_DATABASE
+
+    local init_env_query=( )
+
+    local create_user_query="CREATE USER $COCKROACH_USER"
+
+    # If COCKROACH_PASSWORD is not empty, create this default user with given password.
+    if [[ ! -z "$COCKROACH_PASSWORD" ]]; then
+          create_user_query+="WITH PASSWORD $COCKROACH_PASSWORD"
+    fi
+
+    init_env_query+=( -e "$create_user_query" \
+                      -e "GRANT ALL ON DATABASE $COCKROACH_DATABASE TO $COCKROACH_USER" \
+                      -e "GRANT admin TO $COCKROACH_USER " )
+
+    # Check if the given command contains the `--insecure` flag.
+    # If not, add the `--certs-dir=certs` to init_env_query.
+    if [[ "$*" =~ $INSECURE_MODE_REGEX ]]; then
+              init_env_query+=( --insecure )
+              else
+              init_env_query+=( "--certs-dir=$CERTS_DIR" )
+    fi
+
+    run_sql_query "${init_env_query[@]}"
+}
+
+# start_init_node is to start the single node for the initialization.
+start_init_node() {
+  echo "Starting node for the initialization process. This could take couple seconds"
+  rm -f server_fifo; mkfifo server_fifo
+  local start_node_query=( $COCKROACH_ENTRYPOINT start-single-node \
+                           --background --listening-url-file=server_fifo \
+                           --pid-file=server_pid )
+
+  # Check if the given command contains the `--insecure` flag.
+  # If not, add the `--certs-dir=certs` to init_env_query.
+  if [[ "$*" =~ $INSECURE_MODE_REGEX ]]; then
+    start_node_query+=( --insecure )
+    else
+    start_node_query+=( "--certs-dir=$CERTS_DIR" )
+  fi
+
+  # Get the given listening address.
+  [[ "$*" =~ $LISTEN_ADDR_REGEX ]] && start_node_query+=( $BASH_REMATCH )
+
+  # Start the node and run in the background.
+  "${start_node_query[@]}" &
+
+  # Set a 5-minute timeout for the single node starting up.
+  timeout 3000s cat server_fifo>server.url
+  exit_status=$?
+  if [[ $exit_status -eq 124 ]]; then
+      echo >&2 "error: timeout for cockroach init process"
+      exit 1
+  fi
+}
+
+# stop_init_node is to stop the single node for the initialization.
+stop_init_node() {
+    kill $(cat server_pid)
+    while ! kill -0 $(cat server_pid); do
+        echo "finishing cockroach init process"
+        sleep 3
```

Comment:
does this loop need a timeout? see https://stackoverflow.com/a/12321815 for an example (e.g. quit after say 5mins).